discovery/azure/azure.go

```diff
@@ -202,17 +205,27 @@ func createAzureClient(cfg SDConfig) (azureClient, error) {
 
 	bearerAuthorizer := autorest.NewBearerAuthorizer(spt)
 
+	sender := autorest.DecorateSender(&http.Client{
```

Comment:
Does it work with an empty url?

Reply:
I believe so.

https://github.com/prometheus/common/blob/8d1c9f84e3f78cb628a20f8e7be531c508237848/config/http_config.go#L372
https://github.com/hashicorp/terraform-provider-azurerm/blob/2bb3ebd2b33ad5fc37b0eb77979d2d22a94de1b3/azurerm/config.go#L334